expat/lib/xmlparse.c

```diff
@@ -6295,6 +6295,30 @@ poolStoreString(STRING_POOL *pool, const ENCODING *enc,
   return pool->start;
 }
 
+static size_t
+poolBytesToAllocateFor(int blockSize)
+{
+  /* Unprotected math would be:
+  ** return offsetof(BLOCK, s) + blockSize * sizeof(XML_Char);
+  **
+  ** Detect overflow, avoiding _signed_ overflow undefined behavior
+  ** For a + b * c we check b * c in isolation first, so that addition of a
+  ** on top has no chance of making us accept a small non-negative number
+  */
+  const int stretchedBlockSize = (int)((unsigned)blockSize * sizeof(XML_Char));
```

Comment:
What values of `blockSize` and `sizeof(XML_Char)` is this supposed to work for?
If `sizeof(XML_Char)` can be 4 and `blockSize` can be 2^30 + 1 (that is, 0x40000001), then `stretchedBlockSize` will be set to 4, a positive value, despite an overflow having occurred.